src/Workspaces/VisualBasic/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.vb

```diff
@@ -189,6 +248,45 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Classification.Classifiers
                       ClassificationTypeNames.MethodName)
         End Function
 
+        Private Function TryGetSymbol(
+            node As NameSyntax,
+            symbolInfo As SymbolInfo,
+            semanticModel As SemanticModel) As ISymbol
+
+            Dim symbol = symbolInfo.Symbol
+
+            If symbol Is Nothing AndAlso symbolInfo.CandidateSymbols.Length > 0 Then
+                Dim firstSymbol = symbolInfo.CandidateSymbols(0)
+                Select Case symbolInfo.CandidateReason
+                    Case CandidateReason.NotCreatable
+                        ' Not creatable types are still classified as types.
+                        If firstSymbol.IsConstructor() OrElse TypeOf firstSymbol Is ITypeSymbol Then
+                            symbol = firstSymbol
+                        End If
+
+                    Case CandidateReason.OverloadResolutionFailure
+                        ' If we couldn't bind to a constructor, still classify the type.
+                        If firstSymbol.IsConstructor() Then
+                            symbol = firstSymbol
+                        End If
+
+                    Case CandidateReason.Inaccessible
+                        ' If we couldn't bind to a constructor, still classify the type if its accessible
+                        If firstSymbol.IsConstructor() AndAlso semanticModel.IsAccessible(node.SpanStart, firstSymbol.ContainingType) Then
+                            symbol = firstSymbol
+                        End If
+                End Select
+            End If
```

Comment:
interesting.  does C# have this logic?  can we share it in a cmmon place?  it doesn't seem particularly specific to VB.